core/modules/filter/filter.pages.inc

```diff
@@ -77,6 +78,72 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
 }
 
 /**
+ * Autocomplete callback for link textfield.
+ */
+function _filter_link_autocomplete($string) {
+  $matches = array();
+  if ($string) {
+    // Get node matches.
+    $results = db_select('node')
+      ->fields('node', array('nid', 'title'))
+      ->condition('title', db_like($string) . '%', 'LIKE')
+      ->addTag('node_access')
+      ->range(0, 10)
+      ->execute();
+    foreach ($results as $result) {
+      $node = node_load($result->nid);
```

Comment:
Hm, could we avoid the node_load() here? We already have the NID and Title. Are we getting anything by also loading?